resources/python/pipreqs.py

```diff
@@ -51,8 +51,27 @@ def get_all_imports(
             with open_func(file_name, "r", encoding=encoding) as f:
                 contents = f.read()
             try:
+                # We need to be able to reference a pragma in the comments
+                lines = contents.split('\n')
                 tree = ast.parse(contents)
                 for node in ast.walk(tree):
+                    try:
```

Comment:
doing this reconstruction on every single node in the AST sound expensive. would it be easy to reconstruct only when the `isinstance(node, ast.Import)` or `isinstance(node, ast.ImportFrom)` are hit. Then we can just tag the import with extra details from the pragma.